src/utils/useOutsideClick.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect } from 'react';
+import noop from 'lodash/noop';
+
+// Utilizes the useEffect hook to handle when a click event is fired on the document whether the click event
+// occurred outside of the provided element reference
+export default function useOutsideClick(ref: React.RefObject<Element>, callback: () => void = noop): void {
+    useEffect(() => {
```

Comment:
We've been using the dot-notation for all React methods elsewhere, so `React.useEffect`.